pkg/networkservice/common/authorize/server.go

```diff
@@ -28,19 +31,55 @@ import (
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
 )
 
-type authorizeServer struct{}
+type authorizeServer struct {
+	policy *rego.PreparedEvalQuery
+}
 
 // NewServer - returns a new authorization networkservicemesh.NetworkServiceServers
-func NewServer() networkservice.NetworkServiceServer {
-	return &authorizeServer{}
+func NewServer(p *rego.PreparedEvalQuery) networkservice.NetworkServiceServer {
```

Comment:
How would we update the policy after startup time?

Reply:
This is interesting: https://www.openpolicyagent.org/docs/latest/management/

Reply:
For a future patch, it would be useful to add support for requesting an eval from an OPA microservice rather than running a local eval. This is beneficial because the OPA microservice may make a decision based using access to a store or database which the NSE shouldn't have access to.

Reply:
Added #52 to cover this.